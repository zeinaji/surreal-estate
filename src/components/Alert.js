import React from "react";
import PropTypes from "prop-types";
import "../styles/Alert.css";

const Alert = ({ message, success }) => {
  return (
    <div className={`alert ${success ? "success" : "fail"}`}>{message}</div>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
