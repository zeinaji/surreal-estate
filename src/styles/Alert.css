@import url("https://fonts.googleapis.com/css2?family=Nunito&display=swap");

.alert {
  margin: 5px 0px 20px 0px;
  padding: 3px 10px 3px 10px;
  font-family: "Nunito", sans-serif;
  float: left;
  border-radius: 10px;
  width: 400px;
}

.success {
  background-color: rgb(104, 197, 143);
}

.fail {
  background-color: rgb(235, 112, 112);
}
