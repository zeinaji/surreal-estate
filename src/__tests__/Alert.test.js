import React from "react";
import Alert from "../components/Alert";
import { render } from "@testing-library/react";

describe("Alert", () => {
  xit("it renders", () => {
    const { asFragment } = render(<Alert message="Error!" success={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error message", () => {
    const { getByText } = render(<Alert message="Error!" success={false} />);
    expect(getByText("Error!")).toBeInTheDocument();
  });

  it("renders success message", () => {
    const { getByText } = render(<Alert message="Success!!!" success={true} />);
    expect(getByText("Success!!!")).toBeInTheDocument();
  });
});
