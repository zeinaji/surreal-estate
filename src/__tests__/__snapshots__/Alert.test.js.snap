// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert it renders 1`] = `
<DocumentFragment>
  <div
    class="alert fail"
  >
    Error!
  </div>
</DocumentFragment>
`;
